### ISSUE TITLE
Make hot-deployment only process python modules that have changed

### ISSUE THREAD

**Issue by jjmurre** [zatosource/zato#895]
Zato stores a SHA-256 hash in the database of a python module that is being hot-deployed.

At this moment, the hot-deployed module is always processed. By taking the SHA into account, only modules with changes could be processed. 

This makes it easier to just deploy a set of services modules, regardless of changes since the last deployment of these modules (e.g. by using git pull).

**Comment by dsuch** [comment]
Hi @jjmurre,

I set out to do it but I realized that it may have unforeseen consequences in a rare but possible case of:

* A package is saved to SQL DB
* A notification is about to be sent to servers about the new package
* TCP connection between servers is lost for any reason

Now, users will never be able to redeploy the package because it has been deployed already and its hashsum matches what someone is trying to deploy again.

I am aware of environments that use git pull to deploy new or changed code and it never really matters that some of it did not change. Will that not continue to work for you?
